nncf/common/accuracy_aware_training/training_loop.py

```diff
@@ -209,6 +209,9 @@ def __init__(self,
             raise RuntimeError('No compression algorithm supported by the accuracy-aware training '
                                'runner was specified in the config')
 
+        if maximal_compression_rate > self.adaptive_controller.maximal_compression_rate:
+            maximal_compression_rate = self.adaptive_controller.maximal_compression_rate
```

Comment:
Just to clarify, AdaptiveCompressiontrainingLoop is ablle to reach the `maximal_compression_rate `?

Reply:
If model compresses successfully, it will reach `(maximal_compression_rate - compression_rate_step)` compression rate, where `compression_rate_step` is the last compression rate change step of the algo. For example, if the compression rate is only increasing then the algorithm will exit at `(maximal_compression_rate - 0.1)` compression rate by default.